File: frontend/src/types/member.ts
export interface Member {
  id: string;
  name: string;
  firstName: string;
  lastName: string;
  otherNames?: string;
  emailAddress?: string;
  phoneNumber: string;
  dateOfBirth: string; // ISO format YYYY-MM-DD
  joinedDate?: string;
  gender: 'Male' | 'Female' | 'Other';
  residentialAddress: string;
  city: string;
  hometown: string;
  region: string;
  gpsAddress?: string;
  maritalStatus: 'Single' | 'Married' | 'Widowed';
  nextOfKin: string;
  emergencyContactName: string;
  emergencyContactPhoneNumber: string;
  status: 'Active' | 'Inactive' | 'Archived' | 'Transferred';
  createdAt: string;
}

export interface PagedResponse<T> {
  data: T[];
  hasMore: boolean;
  nextCursor: string | null;
}

export interface MemberFilters {
  name?: string;
  status?: string;
  gender?: string;
  joinedDate?: string;
}

File: backend/src/Echo.Domain/Members/Member.cs
using Echo.Domain.Congregations;

namespace Echo.Domain.Members;

public class Member : ISearchable, ISoftDeletable
{
    public Guid Id { get; set; }
    public Guid CongregationId { get; set; }
    public Congregation Congregation { get; set; } = null!;

    public string Name { get; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string? OtherNames { get; set; } = string.Empty;
    public string? EmailAddress { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public DateOnly DateOfBirth { get; set; }
    public DateOnly? JoinedDate { get; set; }
    public Gender Gender { get; set; }
    public string ResidentialAddress { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Hometown { get; set; } = string.Empty;
    public Region Region { get; set; }
    public string? GpsAddress { get; set; } = string.Empty;
    public MaritalStatus MaritalStatus { get; set; }
    public string NextOfKin { get; set; } = string.Empty;
    public string EmergencyContactName { get; set; } = string.Empty;
    public string EmergencyContactPhoneNumber { get; set; } = string.Empty;
    public MemberStatus Status { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }
}
